### PR TITLE
Build: Retry SourceForge (NSIS) downloads even more

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**
As Windows builds are [still failing](https://github.com/jamulussoftware/jamulus/runs/7982213149?check_suite_focus=true#step:10:30) due to SourceForge/NSIS downloads, this PR ups the number of attempts to 10 and increases the wait time between attempts.

CHANGELOG: SKIP

**Context: Fixes an issue?**
Failing Windows Autobuilds (no Github Issue for that).
Related: #2775

**Does this change need documentation? What needs to be documented and how?**
No.

**Status of this Pull Request**
Ready.

**What is missing until this pull request can be merged?**
Reviews.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: windows/deploy_windows.ps1 [hunk 1]
```diff
@@ -72,13 +73,13 @@ Function Get-RedirectedUrl {
             $response.Close()
             return
         } catch {
-            if ($attempt -lt 3) {
+            if ($attempt -lt $num_attempts) {
                 Write-Warning "Caught error: $_"
-                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt} for $url failed, trying again in 10s"
-                Start-Sleep -Seconds 10
+                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt}/${num_attempts} for $url failed, trying again in 60s"
+                Start-Sleep -Seconds 60
```
Comment (ann0see) [inline windows/deploy_windows.ps1:79]: I'd do a bounded exponential wait here 10,20,40,80,80,80,...

**Review by ann0see — COMMENTED** [review]
Thanks.

File: windows/deploy_windows.ps1 (hunk 1 above)
Comment (hoffie) [reply windows/deploy_windows.ps1]: Done.

**Review by ann0see — APPROVED** [review]
Great. Thanks

**Comment by hoffie** [comment]
Logs show it working properly as well: https://github.com/jamulussoftware/jamulus/runs/7983417675?check_suite_focus=true#step:10:22

File: windows/deploy_windows.ps1 [hunk 2]
```diff
@@ -72,13 +74,14 @@ Function Get-RedirectedUrl {
...
             return
         } catch {
-            if ($attempt -lt 3) {
+            if ($attempt -lt $numAttempts) {
                 Write-Warning "Caught error: $_"
-                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt} for $url failed, trying again in 10s"
-                Start-Sleep -Seconds 10
+                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt}/${numAttempts} for $url failed, trying again in ${sleepTime}s"
+                Start-Sleep -Seconds $sleepTime
+                $sleepTime = [Math]::Min($sleepTime * 2, 80);
```
Comment (pljones) [inline windows/deploy_windows.ps1:81]: Go on, put the max into a variable, too.  You know it makes sense...

Comment (hoffie) [reply windows/deploy_windows.ps1]: Done.

At some point, all resilient systems seem to consist of huge a collection of workarounds and retries...

**Review by pljones — APPROVED** [review]

**Pull request merged**